File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3034_number-of-subarrays-that-match-a-pattern-i/solutions/python3/solution.py
from typing import List

class Solution:
    def countMatchingSubarrays(self, nums: List[int], pattern: List[int]) -> int:
        n = len(nums)
        m = len(pattern)
        count = 0
        
        for i in range(n - m):
            match = True
            for k in range(m):
                if (pattern[k] == 1 and not (nums[i + k] < nums[i + k + 1])) or \
                   (pattern[k] == 0 and not (nums[i + k] == nums[i + k + 1])) or \
                   (pattern[k] == -1 and not (nums[i + k] > nums[i + k + 1])):
                    match = False
                    break
            if match:
                count += 1
        
        return count

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3034_number-of-subarrays-that-match-a-pattern-i/solutions/java/solution.java
class Solution {
    public int countMatchingSubarrays(int[] nums, int[] pattern) {
        int count = 0;
        int m = pattern.length;
        int n = nums.length;

        // We are checking subarrays of size m+1 in nums
        for (int i = 0; i <= n - (m + 1); i++) {
            boolean matches = true;
            for (int k = 0; k < m; k++) {
                if ((pattern[k] == 1 && !(nums[i + k + 1] > nums[i + k])) ||
                    (pattern[k] == 0 && !(nums[i + k + 1] == nums[i + k])) ||
                    (pattern[k] == -1 && !(nums[i + k + 1] < nums[i + k]))) {
                    matches = false;
                    break;
                }
            }
            if (matches) {
                count++;
            }
        }
        return count;
    }
}
